invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessUsersGroups/AccessUsersGroups.js

```diff
@@ -0,0 +1,123 @@
+// This file is part of InvenioRDM
+// Copyright (C) 2023-2024 CERN.
+//
+// Invenio RDM Records is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+import _isEmpty from "lodash/isEmpty";
+import { UserGroupAccessSearchResult } from "./UserGroupAccessSearchResult";
+import React, { Component } from "react";
+import { Modal, Loader, Container } from "semantic-ui-react";
+import PropTypes from "prop-types";
+import { ErrorMessage } from "react-invenio-forms";
+import { i18next } from "@translations/invenio_app_rdm/i18next";
+
+export class AccessUsersGroups extends Component {
+  componentDidMount() {
+    const { fetchUsersGroups, endpoint, searchType } = this.props;
+    fetchUsersGroups(`${endpoint}?expand=true`, searchType);
+  }
+
+  componentWillUnmount() {
+    this.cancellableAction && this.cancellableAction.cancel();
+  }
+
+  render() {
+    const {
+      record,
+      permissions,
+      emptyResultText,
+      tableHeaderText,
+      addButtonText,
+      endpoint,
+      searchBarTitle,
+      selectedItemsHeader,
+      fetchMembers,
+      searchType,
+      searchBarTooltip,
+      searchBarPlaceholder,
+      doneButtonTipType,
+      results,
+      loading,
+      error,
+      addedOrDeletedCallback,
+      updatedCallback,
+    } = this.props;
+
+    return (
+      <>
+        {error && (
+          <ErrorMessage
+            header={i18next.t("Something went wrong")}
+            content={error?.response?.data?.message || error.message}
+            icon="exclamation"
+            className="text-align-left"
+            negative
+            size="mini"
+          />
+        )}
+        <Modal.Content className="share-content">
+          {loading && <Loader isLoading active />}
+          {!loading && results !== undefined && (
+            <UserGroupAccessSearchResult
```

Comment:
understanding just from props, I have a feeling that this component has too many responsibilities, could it be simplified and divided to several components?
when we have too many responsibilities on one component, it becomes impossible to decouple functionalities and reuse the code. 
Some of the props are redundant, for example `record` and `recOwner`, you can retrieve the owner inside the component using record. callbackEndpoint and endpoint would be potentially another example.
some of the other props are just passed down from the parent component to UserGroupAccessSearchResult, maybe that would be an indicator to split this component to couple of new ones?